File: nodejs/english-demo/util/xputil.js

var crypto = require('crypto');

/**
 * hash sign
 * @param signSource - the string for sign
 * @return
 */
function sign(signSource) {
    signature = crypto.createHash('sha1').update(signSource).digest('hex');

    return signature;
}   

/**
 * get millisecond
*/
function getMillisecond() {
    return new Date().getTime();
} 

/**
 * generate repeated string
 * @param str - the source string
 * @param repeatTimes - repeated times
*/
function strRepeat(str, repeatTimes) {
    len = repeatTimes + 1  
    return new Array(len).join(str);
}

module.exports = {sign, getMillisecond, strRepeat};

File: python/chinese-demo/xpyunopensdk/util/xputil.py
"""
 哈稀签名
 @param source - 源字符串
 @return
"""
import hashlib
import time


def sign(source: str):
    sha = hashlib.sha1(source.encode('utf-8'))
    signature = sha.hexdigest()
    return signature


# 获得毫秒数
def getMillisecond():
    t = time.time()
    return int(round(t * 1000))

# 字符串重复多次
def strRepeat(str, times):
    return str * times
